sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -114,16 +114,27 @@ void gnrc_ipv6_nib_init_iface(gnrc_netif_t *netif)
 #if GNRC_IPV6_NIB_CONF_6LN
     netif->ipv6.rs_sent = 0;
 #endif  /* GNRC_IPV6_NIB_CONF_6LN */
-    if (netif->device_type == NETDEV_TYPE_IEEE802154) {
-        /* see https://tools.ietf.org/html/rfc6775#section-5.2 */
-        uint16_t src_len = IEEE802154_LONG_ADDRESS_LEN;
-        gnrc_netapi_opt_t opt = { .opt = NETOPT_SRC_LEN,
-                                  .data = &src_len,
-                                  .data_len = sizeof(src_len) };
-
-        /* XXX we are supposed to be in interface context here, so use driver
-         * directly everything else would deadlock anyway */
-        netif->ops->set(netif, &opt);
+    switch (netif->device_type) {
+        case NETDEV_TYPE_IEEE802154: {
+            /* see https://tools.ietf.org/html/rfc6775#section-5.2 */
+            uint16_t src_len = IEEE802154_LONG_ADDRESS_LEN;
+            gnrc_netapi_opt_t opt = { .opt = NETOPT_SRC_LEN,
+                                      .data = &src_len,
+                                      .data_len = sizeof(src_len) };
+
+            /* XXX we are supposed to be in interface context here, so use driver
+             * directly everything else would deadlock anyway */
+            netif->ops->set(netif, &opt);
+            /* intentionally falls through */
+        }
+        case NETDEV_TYPE_BLE:
+        case NETDEV_TYPE_NRFMIN:
```

Comment:
@gschorcht note that I intentionally left out `NETDEV_TYPE_ESP_NOW` here for now. After reading your comments in #10524: should I add it?

Reply:
Ping? Hope this didn't get lost.

Reply:
(though I moved the function now)

Reply:
I'm still rolling through the changes and discussions. I'm a bit slower than you :wink:

Reply:
> I'm still rolling through the changes and discussions. I'm a bit slower than you

In some cases this could be better (see all my "oops"s and "argh"s ;-D)

Reply:
Ok for the moment. Once #10723 is solved, there should be be no reason why ESP-NOW nodes shouldn't talk 6Lo-ND. It worked in the tests with the workaround as expected.

There might be another reason why ESP-NOW nodes are different, the RSSI. It is not possible to extract the RSSI for a single message. RSSI is only measured during the scan-for-peers phase every 10 s. On the other hand, I did not find any place in the source code, where RSSI  or LQI are used for routing or something like that.;

Reply:
> There might be another reason why ESP-NOW nodes are different, the RSSI. It is not possible to extract the RSSI for a single message. RSSI is only measured during the scan-for-peers phase every 10 s. On the other hand, I did not find any place in the source code, where RSSI or LQI are used for routing or something like that.;

Yupp, neither RSSI nor LQI are of interest for neighbor discovery, just for routing protocols (but AFAIK we don't use it as a metric for RPL at the moment).